File: frontend/src/types/Account.ts
import { Agency } from "./Agency";
import { Customer } from "./Customer";
import { Employee } from "./Employee";
export interface Account {
  id?: string;
  account?: string;
  password?: string;
  balance?: number;
  employee?: Employee;
  agency?: Agency;
  customer?: Customer;
}

export interface AccountInterface {
  getAllByCustomerId(id: string): Promise<Account[]>;
  findByAgencyAndAccount(data: Account): Promise<Account>;
  getById(id: string): Promise<Account>;
  transferBalance(
    id1: string,
    id2: string,
    balance: number,
  ): Promise<Account>;
}

File: back-end-java/src/main/java/com/org/back_end_java/infra/entity/Account.java
package com.org.back_end_java.infra.entity;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "account_tb")
public class Account implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	private String id;
	@Column(nullable = false, unique = true)
	private String number;
	@Column(nullable = false)
	private BigDecimal balance;
	@Column(nullable = false)
	@Enumerated(EnumType.STRING)
	private AccountType accountType;
	@Column(nullable = false, unique = true)
	private String password;
	@ManyToOne
	@JoinColumn(name = "customer_id", nullable = false)
	private Customer customer;
	@ManyToOne
	@JoinColumn(name = "agency_id", nullable = false)
	private Agency agency;
}
